pkg/kubelet/volumes.go

```diff
@@ -193,14 +192,22 @@ func (kl *Kubelet) getPodVolumes(podUID types.UID) ([]*volumeTuple, error) {
 	return volumes, nil
 }
 
+// cleanerTuple is a union struct to allow separating detaching from the cleaner.
+// some volumes require detachment but not all.  Cleaner cannot be nil but Detacher is optional.
+type cleanerTuple struct {
+	Cleaner  volume.Cleaner
```

Comment:
I'm not sure if "tuple" is the right noun for this object. For clarity, how about calling this struct just `cleaner`, and rename `volume.Cleaner` to `volume.Unmounter`?